File: openapi-generator/dart-jaguar/lib/model/dvir_base_vehicle.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';

part 'dvir_base_vehicle.jser.dart';

class DvirBaseVehicle {
   /* The vehicle on which DVIR was done. */
  @Alias('name')
  final String name;
   /* The vehicle id on which DVIR was done. */
  @Alias('id')
  final int id;
  

  DvirBaseVehicle(
      

{
     this.name = null,  
     this.id = null 
    
    }
  );

  @override
  String toString() {
    return 'DvirBaseVehicle[name=$name, id=$id, ]';
  }
}

@GenSerializer(nullableFields: true)
class DvirBaseVehicleSerializer extends Serializer<DvirBaseVehicle> with _$DvirBaseVehicleSerializer {

}

File: openapi-generator/scalaz/src/main/scala/org/openapitools/client/api/DvirBaseVehicle.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import DvirBaseVehicle._

case class DvirBaseVehicle (
  /* The vehicle on which DVIR was done. */
  name: Option[String],
/* The vehicle id on which DVIR was done. */
  id: Option[Long])

object DvirBaseVehicle {
  import DateTimeCodecs._

  implicit val DvirBaseVehicleCodecJson: CodecJson[DvirBaseVehicle] = CodecJson.derive[DvirBaseVehicle]
  implicit val DvirBaseVehicleDecoder: EntityDecoder[DvirBaseVehicle] = jsonOf[DvirBaseVehicle]
  implicit val DvirBaseVehicleEncoder: EntityEncoder[DvirBaseVehicle] = jsonEncoderOf[DvirBaseVehicle]
}
